packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -147,17 +147,23 @@ export class StripeWebhookHandler extends StripeHandler {
       'invoices'
     );
 
-    // This invoice must be for a paypal subscription for us to issue a
-    // paypal refund and be marked for out_of_band refund.
-    if (
-      invoice.collection_method !== 'send_invoice' ||
-      !creditNote.out_of_band_amount ||
-      creditNote.out_of_band_amount === 0
-    ) {
-      reportSentryError(
-        new Error(`Credit note issued for account balance: ${creditNote.id}`),
-        request
-      );
+    if (invoice.collection_method === 'charge_automatically') {
+      // This is a Stripe charge, report if needed and return as Stripe handles
+      // refunding the customer.
+      if (
+        creditNote.customer_balance_transaction ||
```

Comment:
Wouldn't we still want to check for an account balance for PayPal customers and report an error? That would mean it was issued incorrectly, right?

Admittedly, I guess we weren't checking for that here originally for either Stripe or PayPal.

Reply:
This only exists if the credit note adjusts the account balance, which only happens if the user is being credited to their balance which we currently don't want.

Reply:
Hmm yeah I understand that part, but we aren't checking if the credit note adjusts the account balance for PayPal customers with this logic -- only for Stripe customers. Shouldn't we also check this for PayPal customers?

Reply:
We will want that in a future PR to allow partial paypal refunds. There's a if block farther down here that ensures the out_of_band amount perfectly matches the invoice amount. If there was a account balance change for a PayPal user, then that check below will fail and it will error.